apps/condo/domains/property/components/panels/Builder/MapConstructor.ts

```diff
@@ -150,24 +150,30 @@ class Map {
         }
         return true
     }
-
-    public validateUniqueUnitLabel (renamedUnit: string = null, destination: keyof typeof MapViewMode = null): boolean {
-        const unitLabels = this.map.sections
+    get getUnitLabelsWithoutPreview (): string[] {
+        return this.map.sections
             ?.map((section) => section.floors
                 ?.map(floor => floor.units
-                    ?.map(unit => unit.label)
+                    ?.map(unit => !unit.preview && unit.label)
```

Comment:
Don't you need to filter for 'false' values in case of `unit.preview` will be `true`?

Reply:
Only units without `preview` status are needed here. `map()` will push a unit into the array if it is not a `preview`

Reply:
I mean following:

```
const a = [
    {name: 'Bob', preview: true},
    {name: 'Alice', preview: false}
  ]
  const b = a.map(p => !p.preview && p.name)
  console.debug('b', b)
```

You will get `[false, "Alice"]`.

Do you really need `false` item?

Reply:
ah, got it :) Actually, this will not affect the work, but I think it's better to fix it